data_structures/src/chain/mod.rs

```diff
@@ -311,9 +311,14 @@ impl ConsensusConstantsWit2 {
     }
 
     /// Replication factor for data request solving
-    pub fn get_replication_factor(self, epoch: Epoch) -> u16 {
+    pub fn get_replication_factor(self, epoch: Epoch, prev_epoch: Epoch) -> u16 {
         if get_protocol_version(Some(epoch)) >= ProtocolVersion::V2_0 {
-            4
+            // linearly increasing replication factor
+            return if epoch >= prev_epoch {
+                u16::try_from(12 * (epoch - prev_epoch)).unwrap_or(u16::MAX)
```

Comment:
The higher the growing factor, the less probable that a big enough group of malicious stake entries can collude to temporarily censor the chain. For taking over up to 10 consecutive blocks, an attacker would need to place 120 stake entries on the top ranking, simultaneously. Depending on the actual census, and whether multiple stake entries per validator address is enabled or not, this number could either be considered relatively high, or most probably (in the long run, if we expect census to grow in time), relatively low. 

Nevertheless, the replication factor is proposed to be capped to some proportion of the total census. Two thirds has been initially proposed under this rationale: "if more than two thirds of census decide to censor the chain, the chain is to be censored". 

In terms of P2P network traffic, one could argue that up to 2/3rds of eligible block proposers may potentially flood the network with competing block candidates, but not really:
- Nodes should only propagate block candidates only if better than the best validated so far.
- On wit/2.1+, eligible candidates will only be able to propose one single block, no matter how many stake entries they have within the top ranking.

Reply:
I'm okay with the capping, but was thinking that it would be better to follow the same power law as with data request eligibility: `4 * 2 ** (epoch - prev_epoch - 1)`.

epoch | power | linear
--|--|--
1 | 4 | 12
2 | 8 | 24
3 | 16 | 36
4 | 32 | 48
5 | 64 | 60
6 | 128 | 72
7 | 256 | 84
8 | 512 | 96
9 | 1024 | 108
10 | 2048 | 120




In the end it's unlikely to really matter though, both for the ratio of blocks mined and actual liveness.

Reply:
That's my line of thinking too. When assessing the right replication factor _RF_, we need to take into account how credibly can an attacker take over the top _RF_ positions in the rank. Under consideration of the economic criteria presented by @drcpu-github above, and based on what we've seen in the 4 years of mainnet, it is reasonable to think that 12 may be a bit too much for a start.

On the other hand, what's the downside of a high-ish initial RF value? Bloating the network with redundant messages doesn't sound specially of concern here, under the assumption that block candidates are still subject to comparison and filtering as they travel through the network.

As per the _power_ vs. _linear_ progression, I favor _power_ because it has worked pretty decently so far, but don't oppose to _linear_ if it is found to be more suitable for PoS.